src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1206,6 +1206,19 @@ class LimitClauseSegment(BaseSegment):
     )
 
 
+@spark3_dialect.segment(replace=True)
+class SetOperatorSegment(BaseSegment):
+    """A set operator such as Union, Minus, Except or Intersect."""
+
+    type = "set_operator"
+    # https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-setops.html
+    match_grammar = OneOf(
+        Sequence(OneOf("EXCEPT", "MINUS"), Ref.keyword("ALL", optional=True)),
+        Sequence(OneOf("UNION", "INTERSECT"), OneOf("DISTINCT", "ALL", optional=True)),
+        exclude=Sequence("EXCEPT", Bracketed(Anything())),
```

Comment:
Does this line work and is it needed a ince the brackets won’t be part of this segment?

I don’t think any of the test cases cover it?